community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/slurmsync.py

```diff
@@ -54,21 +54,91 @@
 TOT_REQ_CNT = 1000
 _MAINTENANCE_SBATCH_SCRIPT_PATH = dirs.custom_scripts / "perform_maintenance.sh"
 
-NodeStatus = Enum(
-    "NodeStatus",
-    (
-        "orphan",
-        "power_down",
-        "preempted",
-        "restore",
-        "resume",
-        "terminated",
-        "unbacked",
-        "unchanged",
-        "unknown",
-    ),
-)
+class NodeAction(Protocol):
+    def apply(self, nodes: List[str], hostlist: List[str], count: int) -> None:
```

Comment:
Why `hostlist` is in arguments? 
`hostlist` is not `List[str]`, but `str` and should be produced ad-hoc by using `util.to_hostlist_fast(nodes)`

Reply:
Did it mainly due to repetition and Actions that contain multiple subactions (only `NodeActionPrempt` to avoid calling it `to_hostlist` more than once), but changing the interface to make it nodes only.